File: JavaScript/8js 1/getAge.js
/*
|--------------------------------------------------------------------------
| Parse nice int from char problem
|--------------------------------------------------------------------------
|
*/

function getAge(inputString) {
   return Number(inputString.split(' ')[0]);
}


function getAge(inputString) {
   return parseInt(inputString);
}


function getAge(inputString) {
   return +inputString[0];
}


function getAge(inputString) {
   return inputString.match(/[0-9]/)[0] * 1;
}


console.log(getAge("4 years old"), 4);

File: Python/8py 1/get_age.py
"""
|--------------------------------------------------------------------------
| Parse nice int from char problem
|--------------------------------------------------------------------------
|
"""

def get_age(age):
   return int(age.split(' ')[0])


import re
def get_age(age):
   return int(re.search(r"\d+", age).group())


def get_age(age):
   return int(age[0])


def get_age(age):
   return int(__import__('re').search(r'\d+', age).group(0))


def get_age(age):
   for x in age:
      if x != str(age):
         return int(x)


print(get_age("2 years old"), 2)
print(get_age("4 years old"), 4)
print(get_age("5 years old"), 5)
print(get_age("7 years old"), 7)